user settings: Custom Field followup bugs

YJDave opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Followup of #12753 

**Testing Plan:** <!-- How have you tested? -->


**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->

![user_settings_cp_alert](https://user-images.githubusercontent.com/25907420/62081697-c43e2880-b270-11e9-98fc-9da77c0a875f.gif)

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
I think with the sizing you have there, you could put it in the original location to the right of the label (and then we don't need to deal with the issues of vertical scroll changing)

YJDave commented:
@timabbott  I have updated the first save profile widget style.
The second
![save-user-settings](https://user-images.githubusercontent.com/25907420/62265659-9577ac00-b442-11e9-839a-af0f33b716c9.gif)
 commit is still in WIP.

YJDave commented:
@rishig @timabbott updated. 

Demo:
Save widget style:g
![Screen Shot 2019-08-24 at 11 28 44 PM](https://user-images.githubusercontent.com/25907420/63641130-f3588600-c6c6-11e9-83d0-54d27864960b.png)

Default custom field:
![custom-field-style](https://user-images.githubusercontent.com/25907420/63641099-67def500-c6c6-11e9-8efe-7a7dd9437d7e.gif)

There is minor error with lint, which

timabbott commented on zerver/views/custom_profile_fields.py:
```diff
@@ -57,13 +61,29 @@ def validate_custom_field_data(field_type: int,
+        if name and name != default_field['name']:
+            raise JsonableError(_("Default custom profile field data cannot changed."))
+        if hint and hint != default_field['hint']:
+            raise JsonableError(_("Default custom profile field data cannot changed."))
```
This isn't great English, and I'm also not sure how to briefly explain the user's mistake here.  @rishig do you think we need an error for this case, or should we just ignore whatever was set in `name` and use the value from `default_field`?  This isn't a possible situation from the UI, just the API.

rishig replied:
I don't have a good intuition for the API, but I would say either:
* No error / silently overwrite `name`
* Maybe something like "`name` must be empty for type `external field`"?

timabbott commented on zerver/lib/external_accounts.py:
```diff
@@ -10,12 +10,16 @@
-        "url_pattern": "https://github.com/%(username)s"
+        "url_pattern": "https://github.com/%(username)s",
+        "name": "GitHub",
+        "hint": "Only your GitHub username",
```
@rishig what should the hint be here?  "Enter your GitHub username" sounds good to me.

rishig replied:
yeah, sgtm

timabbott commented on zerver/lib/external_accounts.py:
```diff
@@ -10,12 +10,16 @@
     "twitter": {
-        "text": "Twitter",
-        "url_pattern": "https://twitter.com/%(username)s"
+        "text": "Twitter",  # Realm admin sees in field options
```
Let's move these comments explaining what the values do to a block comment at the top of `DEFAULT_EXTERNAL_ACCOUNTS`

rishig commented:
> I think the right answer might be to move "Field type" and its subvalues above Label and Hint unconditionally; it seems reasonable to ask users to first pick a field type and then fill in other values as needed.

Agreed

YJDave commented:
Thanks for the review!
I have addressed all above comments.
![image](https://user-images.githubusercontent.com/25907420/63825090-faa5bb00-c976-11e9-820a-3e6389fa9e79.png)

I belive we also need add validation to not allow admin to create any type of field with name similar to our default field names. Otherwise, it will throw error if admin will try to create defaut external field after that field is created.
So if admin creates `Text` type field `GitHub` and then try to add `GitHub` external field, it won't allow.

timabbott commented:
I think it's actually fine to allow admins to create a field with the name "GitHub" -- we should just give a good error message when they try to create the external account field with that name.

timabbott commented:
Merged as the series ending with c06d8129a341b849546bbd10c04bb471ec297249, which includes adding a placeholder that should make this actually useable for someone who hasn't seen the implementation :).  

For next steps:
* We should add a quick unit test for the field name reuse issue you described above.
* @rishig I guess we should open an issue describing the plan for using logos for external accounts rather than names?  There's a v1 version of this where you only get the logo for the "built-in" values.  
* Also, want to add a few more default ones to `zerver/lib/external_accounts.py`?
